components/gateway/pkg/proxy/transport.go

```diff
@@ -12,30 +12,44 @@ import (
 	"github.com/form3tech-oss/jwt-go"
 	"github.com/kyma-incubator/compass/components/director/pkg/correlation"
 	"github.com/kyma-incubator/compass/components/gateway/pkg/httpcommon"
-
 	"github.com/pkg/errors"
 )
 
 var emptyQuery error = errors.New("empty graphql query")
 
-//go:generate mockery -name=RoundTrip -output=automock -outpkg=automock -case=underscore
+//go:generate mockery --name=RoundTrip --output=automock --outpkg=automock --case=underscore
 type RoundTrip interface {
 	RoundTrip(*http.Request) (*http.Response, error)
 }
 
-//go:generate mockery -name=AuditlogService -output=automock -outpkg=automock -case=underscore
+//go:generate mockery --name=AuditlogService --output=automock --outpkg=automock --case=underscore
 type AuditlogService interface {
-	Log(ctx context.Context, request, response string, claims Claims) error
+	Log(ctx context.Context, msg AuditlogMessage) error
+}
+
+//go:generate mockery --name=PreAuditlogService --output=automock --outpkg=automock --case=underscore
+type PreAuditlogService interface {
+	AuditlogService
+	PreLog(ctx context.Context, msg AuditlogMessage) error
+}
+
+type AuditlogMessage struct {
+	CorrelationIDHeaders correlation.Headers
```

Comment:
The auditlog message only needs one correlation id, so better provide only a correlation id string

Reply:
Yes, that's true! But we've left the message to carry all of the correlation headers because of the new child context creation in the auditlog queue workers upon calls to auditlog service. This can be seen here: https://github.com/kyma-incubator/compass/pull/1626/files#diff-252a4af9a7b98d32685344f445c56cc7ef76a7c94ca896111f515fbdec3a16afR36